spec/src/main/asciidoc/cdi.asciidoc

```diff
@@ -99,6 +99,29 @@ public class BarBean {
 
 If multiple configurations of `ManagedExecutor` are assigned the same name, the container detects it and raises a `javax.enterprise.inject.spi.DefinitionException`.
 
+==== Programmatic CDI lookup
+
+Injection points that are unqualified (i.e. `@Default`) have a `@NamedInstance("<generated name>")` qualifier added by the container. Therefore, this must be taken into account when performing a programmatic CDI lookup.  For example, given the following injection point:
+
+[source, java]
+----
+package com.foo;
+@ApplicationScoped
+public class FooBean {
+  @Inject
+  ManagedExecutor exec; // has generated name of "com.foo.FooBean.exec"
+}
+----
+
+To perform an equivalent programmatic CDI lookup, the generated `@NamedInstance` qualifier must be taken into account. Namely:
+
+[source, java]
+----
+ManagedExecutor exec = CDI.current().select(ManagedExecutor.class, NamedInstance.Literal.of("com.foo.FooBean.exec")).get();
+----
+
```

Comment:
The above section seems like implementation detail that would create an awkward usage pattern for the user.  Can we omit that from the spec?  And then keep the following section which clarifies the behavior for programmatic lookup of unqualified instance

Reply:
Good point, that will cut down the size of this section considerably, and avoid exposing this possible implementation detail.